### PR TITLE
Fix non deterministic job status spec and make owner deletion an atom…

### PR DESCRIPTION
**Pull request by vritant**
…ic transaction

### CODE REVIEW

**Comment by vritant** [comment]
@mstead FYI

File: server/src/test/java/org/candlepin/resource/OwnerResourceUeberCertOperationsTest.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void setUp() {
             null, null, poolManager, null, null, null,
             null, consumerTypeCurator, entCertCurator, entitlementCurator,
             ueberCertGenerator, null, null, contentOverrideValidator,
-            serviceLevelValidator, null, null, null, null, null);
+                serviceLevelValidator, null, null, null);
```
Comment (wottop) [inline server/src/test/java/org/candlepin/resource/OwnerResourceUeberCertOperationsTest.java]: remove leading tab here

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 2]
```diff
@@ -342,101 +334,21 @@ public Owner createOwner(Owner owner) {
...
-            log.info("Deleting environment: {}", e.getId());
-            envCurator.delete(e);
-        }
-
-        for (Pool p : poolManager.listPoolsByOwner(owner)) {
-            log.info("Deleting pool: {}", p);
-            poolManager.deletePool(p);
-        }
-
-        cleanupUeberCert(owner);
```
Comment (wottop) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: Are we just going to drop this? It looks like the one thing that didn't make it across to the new class

File: server/src/test/java/org/candlepin/resource/OwnerResourceUeberCertOperationsTest.java (hunk 1 above)
Comment (vritant) [inline server/src/test/java/org/candlepin/resource/OwnerResourceUeberCertOperationsTest.java]: @wottop fixed

File: server/src/main/java/org/candlepin/resource/OwnerResource.java (hunk 2 above)
Comment (vritant) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: @wottop this was only a three line method, and the only instance where it was called, so I replaced the contents in OwnerManager, and removed the method.

Comment (wottop) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: ok, I see that

**Comment by wottop** [comment]
When I ran rspec in parallel i got:

Failures:

  1) Consumer Resource updates consumer updated timestamp on bind
     Failure/Error: Unable to find matching line from backtrace
     RestClient::Conflict:
       409 Conflict
     # ./server/client/ruby/candlepin_api.rb:1344:in `delete'
     # ./server/client/ruby/candlepin_api.rb:298:in `delete_owner'

But it doesn't show in serial run. Concern?

**Comment by vritant** [comment]
@wottop definetly a concern. looking.

**Comment by wottop** [comment]
And then this

Failures:

  1) Consumer Resource updates consumer updated timestamp on bind
     Failure/Error: Unable to find matching line from backtrace
     RestClient::Conflict:
       409 Conflict
     # ./server/client/ruby/candlepin_api.rb:1344:in `delete'
     # ./server/client/ruby/candlepin_api.rb:298:in `delete_owner'

**Comment by wottop** [comment]
ANd then this

  1) Uebercert owner can be deleted
     Failure/Error: @cp.delete_owner(owner['key'], false)
     RestClient::InternalServerError:
       500 Internal Server Error
     # ./server/client/ruby/candlepin_api.rb:1344:in `delete'
     # ./server/client/ruby/candlepin_api.rb:298:in `delete_owner'
     # ./server/spec/ueber_cert_spec.rb:19:in `block (2 levels) in <top (required)>'

**Comment by mstead** [comment]
A little more context in the commit history as to why this was needed might be nice in case we had to look back at it.

**Comment by vritant** [comment]
@mstead done in latest version of the branch.

**Comment by wottop** [comment]
ack

**Pull request merged**